clang/lib/Driver/ToolChains/AMDGPU.cpp

```diff
@@ -633,6 +633,17 @@ void amdgpu::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   else if (Args.hasArg(options::OPT_mcpu_EQ))
     CmdArgs.push_back(Args.MakeArgString(
         "-plugin-opt=mcpu=" + Args.getLastArgValue(options::OPT_mcpu_EQ)));
+
```

Comment:
HIPAMD toolchain has its lld command argument handling in AMDGCN::Linker::constructLldCommand. Also need change there.

Reply:
So, I wasn't sure if I should also apply this to HIP / CUDA stuff just yet. HIP doesn't pass LTO bitcode to the linker, so we can't do a full link with the library. CUDA doesn't even invoke the linker either. I think in the future it would be nice to provide these to HIP and CUDA by default however.